ntp-daemon/src/main.rs

```diff
@@ -23,5 +27,66 @@ async fn main() -> Result<(), Box<dyn Error>> {
     // this away when not using sentry.
     let _guard = finish_tracing_init(&mut config, has_log_override)?;
 
-    ntp_daemon::spawn(&config.system, &config.peers).await
+    let peers_reader = Arc::new(tokio::sync::RwLock::new(Vec::new()));
+    let peers_writer = peers_reader.clone();
+
+    std::fs::remove_dir_all("/run/ntpd-rs")?;
+    std::fs::create_dir("/run/ntpd-rs")?;
+
+    let peers_listener = UnixListener::bind("/run/ntpd-rs/observe")?;
+
+    // this binary needs to run as root to be able to adjust the system clock
+    // directories inherit root permissions, but the client should not need elevated
+    // permissions to read from the socket. So we explicitly set the permissions
+    let permissions: std::fs::Permissions = PermissionsExt::from_mode(0o777);
+
+    std::fs::set_permissions("/run/ntpd-rs", permissions.clone())?;
```

Comment:
We may want to make permissions on the directory separately configurable from the socket itself. Also am wondering a bit if creating, then setting permissions isn't a security risk, as it could potentially lead to a permission race on startup.

Reply:
I played around with this a bit, but got nowhere. 

The `bind` call must create the "file". Trying to create the file with the normal rust functions results in a 

```
Error: Os { code: 98, kind: AddrInUse, message: "Address already in use" }
```

when you try `bind` (I made sure the file is closed, so I think it just bases this on the file existing)

Reply:
Perhaps we can avoid this by using umask just before creating the socket.